contracts/modules/STO/USDTiered/USDTieredSTO.sol

```diff
@@ -286,6 +291,7 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
             }
         }
         address walletAddress = (treasuryWallet == address(0) ? IDataStore(getDataStore()).getAddress(TREASURY) : treasuryWallet);
+        // @dev "else" branch is unreachable.
```

Comment:
The treasury address can be set to `0x0` to cover this branch. Similar techniques can be applied to some other unreachable branched. That being said, Unless we are aiming for 100% coverage, I am fine with these branches not being covered.

Reply:
To clarify, my point is, there's no way to have the following `require` statement revert because `walletAddress` cannot be `0x0`, given the conditions above.

Reply:
To clarify, my point is, there's a still a way to trigger the require statement.

The above statement says that 
if treasury wallet is not zero, set wallet address to treasury wallet
but if
treasury wallet  is zero then fetch proper address from the datastore

Now, what if, we set treasury wallet = 0 so that second statement is triggered and we also set the address in datastore to 0? That way the contract will query the datastore but the datastore will return 0x0 so the wallet address will in turn be 0x0.

Reply:
Yeah, but DataStore has it's own checks that prevent setting treasury to 0x0 right?

Reply:
Nope, it's a dumb contract :)

Reply:
There is no explicit check in the dataStore to avoid 0x0 as an address to store. But you can't do it by using change `changeTreasuryWallet()` function of the ST because we have 0x0 check while you have to manage to call `setAddress()` function of the dataStore directly to set globally treasury wallet to 0x0.